### DISCUSSION TITLE
Discontinue an extension and refund / cancel subscriptions

### DISCUSSION THREAD

**Discussion by WofWca** [Glench/ExtPay#225] [Q&A] [answered]
Hello!
We wish to discontinue our extension, and issue a refund to all users whose subscription for the current period is paid.

Or, as a (worse) alternative, cancel all user's subscription's automatic prolongation and only serve for the last paid period, but it's a pain to develop, we 

Is there a standard way to do this? Because it's not possible to achieve this reliably with just JavaScript, because a user might have uninstalled the extension.

Similar (but not the same) discussions:
- https://github.com/Glench/ExtPay/discussions/188
- https://github.com/Glench/ExtPay/discussions/190

**Comment by Glench** [comment, marked as answer]
Hey! I suggest just refunding everyone through Stripe! If you email me I can also just disable access for existing users, but I'd caution against this since they paid for that functionality and if it just stops working suddenly with no explanation in the extension itself then they'll be upset. Instead, I'd just release an update and let the existing users who don't get the update continue on but not pay anything further.

**Reply by WofWca** [reply]
> continue on but not pay anything further

So how can this be done? How can I cancel all subscriptions so that they're not charged anymore, but can continue using the extension for the period that they paid for?

**Reply by WofWca** [reply]
> refunding everyone through Stripe

Are they not going to get charged anymore if I do this?

**Reply by Glench** [reply]
When you cancel a subscription you can just choose the option to stop at the end of the next billing cycle: ![Screenshot 2024-07-16 at 10 07 48 AM](https://github.com/user-attachments/assets/555199ba-91ad-4368-a5dd-a1d808e9fb05)
